grobid-core/src/main/java/org/grobid/core/utilities/GrobidProperties.java

```diff
@@ -688,15 +688,44 @@ public static File getPdfToXMLPath() {
         return pathToPdfToXml;
     }
 
+    private static String getGrobidCRFEngineName(final String modelName) {
+        String defaultEngineName = GrobidProperties.getGrobidCRFEngine().name();
+        // Note: don't use GrobidModels here due to the circular reference
+        if (
+            GrobidProperties.getGrobidCRFEngine() == GrobidCRFEngine.DELFT
+            && (
+                modelName.equals("fulltext")
```

Comment:
It seem to cause some class loading issues when actually running the service (although the tests were fine). That is probably because `GrobidModels` depends on `GrobidProperties` (which then shouldn't depend on `GrobidModels`).